application/dash_application/utility/df_manipulation.py

```diff
@@ -475,3 +475,37 @@ def remove_suffixed_columns(df: DataFrame, suffix: str) -> DataFrame:
     """
     to_drop = [col_name for col_name in df if col_name.endswith(suffix)]
     return df.drop(to_drop, axis=1)
+
+def build_miso_info(df, page_title, metrics):
+    """
+    Builds the JSON for the http body for the request to MISO.
+    Expects an array of dicts in the format:
+    [{
+        'title': threshold name minus the 'maximum' or 'minimum', 
+        'threshold_type': gt, ge, lt, le
+        'threshold': the current threshold setting from the sidebar,
+        'value': the column to check in the dataframe for this threshold
+    }]
+    Returns [json string dump of http body, button style]
+    """
+    if (len(df.index) > 0):
+        miso_button_style = {"display": "inline-block"}
+        miso_request = {'report': page_title, 'library_aliquots': []}
+        for index, row in df.iterrows():
+            split_provenance_id = row[PINERY_COL.SampleProvenanceID].split('_')
+            metrics_this_row = []
+            for metric in metrics:
+                metrics_this_row.append({
+                    'title': metric['title'],
+                    'threshold_type': metric['threshold_type'],
+                    'threshold': metric['threshold'],
+                    'value': row[metric['value']]
+                })
+            miso_request['library_aliquots'].append({
+                'name': split_provenance_id[2],
+                'metrics': metrics_this_row,
+                'run': split_provenance_id[0]
```

Comment:
Would this require `lane` as well?

Reply:
Shouldn't, this is the format @djcooke specified for receiving into MISO.
Dillan, should I be sending MISO the lane as well?

Reply:
Depends how specific you want it to be:

* If you provide only the library aliquot, MISO will include all lanes that it has been run in
* If you provide run + library aliquot, MISO will filter to include only lanes from that run that include the library aliquot
* If you provide run + lane + library aliquot, MISO could filter to include only that lane

If the metrics are specific to a lane, then you probably want to provide the lane.

Reply:
@slazicoicr Metrics are specific to a Lane because of how IUSes work, right?

Reply:
Updated